Updates related to operator status and static pod.

aveshagarwal opened the pull request:
@sjenning @sttts

aveshagarwal commented:
/retest

aveshagarwal commented:
@sttts @sjenning lgtm?

sttts commented on bindata/v3.11.0/kube-scheduler/pod.yaml:
```diff
@@ -5,6 +5,7 @@ metadata:
   namespace: openshift-kube-scheduler
   labels:
     app: openshift-kube-scheduler
+    deployment-id: "DEPLOYMENT_ID"
```
we updated this to REVISION, compare @deads2k's recent library-go PR.

aveshagarwal replied:
@sttts thanks, will update soon.

aveshagarwal commented:
@sttts I have updated it. Right now I have changed this PR to WIP as there are lot of code changes and i want to make sure that tests pass correctly and scheduler works as expected.

aveshagarwal commented:
/retest

aveshagarwal commented:
So i noticed that scheduler pods are coming up fine and other pods are getting scheduled. Monitoring if any failures are related to scheduler.

aveshagarwal commented:
@sttts PTAL

sttts commented:
Static pod changes lgtm.

mfojtik commented:
/lgtm
